### PR TITLE
Feature/maniphest app

### LINKED ISSUE #45: Show a maniphest task
### Proposed Functionality

Implement `phabfive maniphest show T123`

### PR DESCRIPTION
**Pull request by Grokzen**
### Improvements:

Create a new app within phabfive for maniphest. Also added a small helper command for getting into a REPL where access to the API is simpler to reach for testing and debugging of the internals of phabricator API client, very usefull during devleopment.

New supported commands and examples of running them

```
(phabfive) ➜  phabfive git:(feature/maniphest_app) ✗ phabfive maniphest add comment T1 "test from cli"
Comment successfully added              
Ticket URI: https://127.0.0.1/T1                                       
```

```
(phabfive) ➜  phabfive git:(feature/maniphest_app) ✗ phabfive maniphest show T1
Ticket ID:          1
phid:               PHID-TASK-ouldaoojg34napsdneet
authorPHID:         PHID-USER-4m66oxlehcct7p6actyb
ownerPHID:          None
ccPHIDs:            ['PHID-USER-4m66oxlehcct7p6actyb']
status:             open
statusName:         Open
isClosed:           False
priority:           Needs Triage
priorityColor:      violet
title:              gitlab testing thingy...
description:
projectPHIDs:       []
uri:                https://127.0.0.1/T1
auxiliary:          []
objectName:         T1
dateCreated:        1592474276 (2020-06-18 11:57:56)
dateModified:       1592475102 (2020-06-18 12:11:42)
dependsOnTaskPHIDs: []
```

```
(phabfive) ➜  phabfive git:(feature/maniphest_app) phabfive repl
*************
use self.phab to access the phacility API
use self.conf to access current client configuration
use pp() to prettyprint the API response back from self.phab.* calls
*************
--Return--
> /home/grok/dynamist/phabfive/phabfive/repl.py(31)run()->None
-> pdb.set_trace()
(Pdb) 
```

Closes #45

### CODE REVIEW

File: phabfive/cli.py [hunk 1]
```diff
@@ -26,7 +26,9 @@
 Available phabfive commands are:
     passphrase          The passphrase app
     diffusion           The diffusion app
+    maniphest           The maniphest app
     paste               The paste app
+    repl                Enter a REPL where API access is possible
```
Comment (holmboe) [inline phabfive/cli.py:31]:
```suggestion
    repl                Enter a REPL with API access
```

File: phabfive/cli.py [hunk 2]
```diff
@@ -107,6 +109,23 @@
...
+    phabfive repl [options]
+
+Options:
+    -h, --help           Show this help message and exit
+"""
+
+sub_maniphest_args = """
+Usage:
+    phabfive maniphest add comment <ticket_id> <comment> [options]
+    phabfive maniphest info <ticket_id> [options]
```
Comment (holmboe) [inline phabfive/cli.py:123]:
```suggestion
    phabfive maniphest show <ticket_id> [options]
```

File: phabfive/cli.py [hunk 3]
```diff
@@ -107,6 +109,23 @@
...
+Usage:
+    phabfive repl [options]
+
+Options:
+    -h, --help           Show this help message and exit
+"""
+
+sub_maniphest_args = """
+Usage:
+    phabfive maniphest add comment <ticket_id> <comment> [options]
```
Comment (holmboe) [inline phabfive/cli.py:122]:
```suggestion
    phabfive maniphest comment add <ticket_id> <comment> [options]
```

File: phabfive/cli.py [hunk 4]
```diff
@@ -259,6 +283,56 @@ def run(cli_args, sub_args):
...
+                print("dateCreated:        {0} ({1})".format(
+                    result["dateCreated"],
+                    datetime.fromtimestamp(int(result["dateCreated"])),
+                ))
+                print("dateModified:       {0} ({1})".format(
+                    result["dateModified"],
+                    datetime.fromtimestamp(int(result["dateModified"])),
+                ))
+                print("dependsOnTaskPHIDs: {0}".format(result["dependsOnTaskPHIDs"]))
+
```
Comment (holmboe) [inline phabfive/cli.py:335]: Too much info. And too "debuggy" looking. The default output should be more human-friendly, with perhaps only `uri`, `statusName`, and `priority`. On my wishlist is also to list parents and sub-tasks. And the rest of this information should perhaps be hidden behind a `--verbose` flag? IMHO.

Comment (Grokzen) [reply phabfive/cli.py]: This was resolved in the latest commit, the items that was decided to remain is now few, the rest is commented out for future implementation of --verbose flag when bigger rewrite will be done to better support the data manipulation through phabfive.

File: phabfive/cli.py (hunk 3 above)
Comment (Grokzen) [reply phabfive/cli.py]: fixed

File: phabfive/cli.py (hunk 2 above)
Comment (Grokzen) [reply phabfive/cli.py]: fixed

File: phabfive/cli.py (hunk 1 above)
Comment (Grokzen) [reply phabfive/cli.py]: fixed

File: phabfive/cli.py [hunk 5]
```diff
@@ -26,7 +26,9 @@
 Available phabfive commands are:
     passphrase          The passphrase app
     diffusion           The diffusion app
+    maniphest           The maniphest app
     paste               The paste app
+    repl                Enter a REPL with API access is possible
```
Comment (holmboe) [inline phabfive/cli.py:31]:
```suggestion
    repl                Enter a REPL with API access
```

File: phabfive/cli.py [hunk 6]
```diff
@@ -259,6 +285,57 @@ def run(cli_args, sub_args):
...
                 u.print_whoami()
 
+        if cli_args["<command>"] == "repl":
+            r = repl.Repl()
+            r.run()
+
+        if cli_args["<command>"] == "maniphest":
+            m = maniphest.Maniphest()
+
+            if sub_args["add"] and sub_args["comment"]:
```
Comment (holmboe) [inline phabfive/cli.py:295]:
```suggestion
            if sub_args["comment"] and sub_args["add"]:
```

Minor nitpick.

File: phabfive/core.py [hunk 7]
```diff
@@ -50,29 +51,37 @@ def __init__(self):
...
                 if example:
                     error += ", " + example
+
                 raise PhabfiveConfigException(error)
+
         self.phab = Phabricator(
-            host=self.conf.get("PHAB_URL"), token=self.conf.get("PHAB_TOKEN")
+            host=self.conf.get("PHAB_URL"), token=self.conf.get("PHAB_TOKEN"),
         )
+        # This enables extra endpoints that normally is unaccessible
```
Comment (holmboe) [inline phabfive/core.py:83]: To speed things up for the normal use-cases we should probably put this behind some kind of try-except logic that runs `update_interfaces()` only if the endpoint is not already known? Either solve it in this PR, or create a ticket or add a TODO into the code?

**Comment by holmboe** [comment]
LGTM, rebase and merge.

**Pull request merged**